assignment/core/src/main/java/com/assignment/core/workflows/CSVPageCreationWorkflowProcess.java

```diff
@@ -0,0 +1,160 @@
+     *
+     * @param inputStream The InputStream of the CSV file.
+     * @param resourceResolver The resource resolver for page creation.
+     * @throws Exception If an error occurs during CSV processing.
+     */
+    private void processCSV(InputStream inputStream, ResourceResolver resourceResolver) throws IOException, RepositoryException, WCMException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+        String line;
+        boolean isFirstLine = true;
+
+        PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
+        if (pageManager == null) {
+            log.error("PageManager is not available.");
+            return;
+        }
+
+        while ((line = reader.readLine()) != null) {
+            if (isFirstLine) {
+                isFirstLine = false;
+                continue;
+            }
+
+            List<String> values = Arrays.asList(line.split(","));
+            if (values.size() < 5) {
+                log.error("Invalid CSV line, not enough columns: {}", line);
+                continue;
+            }
+
+            String title = values.get(0).trim();
+            String path = values.get(1).trim();
+            String description = values.get(2).trim();
+            String tags = values.get(3).trim();
+            String thumbnail = values.get(4).trim();
+
+            String validPageName = createValidPageName(title);
+
+            createPage(pageManager, path, validPageName, title, description, tags, thumbnail);
+        }
+    }
+
+    /**
+     * Creates a valid page name from the given title.
+     *
+     */
+    private String createValidPageName(String title) {
+        String validName = title.replaceAll("[^\\p{Alnum}_]", "-");
+        return validName.length() > 50 ? validName.substring(0, 50) : validName;
+    }
+
+    /**
+     * Creates a page with the given details.
+     *
+     * @param pageManager  PageManager to create the page.
+     * @param path  path where the page will be created.
+     * @param pageName  name of the page.
+     * @param title  title of the page.
+     * @param description  description of the page.
+     * @param tags  tags for the page.
+     * @param thumbnail thumbnail path for the page from asset.
+     */
+    private void createPage(PageManager pageManager, String path, String pageName, String title, String description, String tags, String thumbnail)
+            throws RepositoryException , WCMException {
+
+        Page page = pageManager.create(path, pageName, "/conf/global/settings/wcm/templates/page", title);
+        if (page != null) {
+            Node contentNode = page.getContentResource().adaptTo(Node.class);
+            if (contentNode != null) {
+                contentNode.setProperty("jcr:description", description);
+                contentNode.setProperty("cq:tags", tags.split(";"));
+                contentNode.setProperty("cq:templateThumbnail", thumbnail);
```

Comment:
cq:templateThumbnail is not the property in which the page's thumbnail image is stored. Use the correct property name, try adding a thumbnail image to the page's properties and check in CRXDE where it is getting stored to find a solution for this.